sources/common/infoschema.go

```diff
@@ -74,7 +74,13 @@ func ProcessSchema(conv *internal.Conv, infoSchema InfoSchema) error {
 // If we can't get/process data for a table, we skip that table and process
 // the remaining tables.
 func ProcessData(conv *internal.Conv, infoSchema InfoSchema) {
-	for srcTable, srcSchema := range conv.SrcSchema {
+	// Tables are ordered in alphabetical order with one exception: interleaved
+	// tables appear after the population of their parent table.
+	orderTableNames := ddl.OrderTable(conv.SpSchema)
+
+	for _, spannerTable := range orderTableNames {
+		srcTable := conv.ToSource[spannerTable].Name
```

Comment:
GetSpannerTable creates a new spanner table name and then assigns the mapping which cannot be done in case of source table name

Reply:
oh ok. Missed that. We should anyways create this method there to fetch the source table. That file should be the only one to directly deal with the mapping (ToSource/ToSpanner). It can also throw explicit errors for input validation and for cases when the mapping/conv is not populated.

Reply:
done